### PR TITLE
#136 implement video source to loop video file

### PR DESCRIPTION
**Pull request by tomskikh**

### CODE REVIEW

File: adapters/gst/gst_plugins/python/media_files_src_bin.py [hunk 1]
```diff
@@ -111,6 +119,10 @@ def do_set_state(self, state: Gst.State):
         if self.current_state == Gst.State.NULL and state != Gst.State.NULL:
             assert self.location is not None, '"location" property is required'
             assert self.file_type is not None, '"file-type" property is required'
+            if self.loop_file:
+                assert (
+                    self.file_type == FileType.VIDEO
+                ), f'Only "fil-type={FileType.VIDEO.value}" is allowed when "loop-file" is enabled'
```
Comment (bwsw) [inline adapters/gst/gst_plugins/python/media_files_src_bin.py:125]: Typo in 'fil-type'

Comment (bwsw) [reply adapters/gst/gst_plugins/python/media_files_src_bin.py:125]: @tomskikh don't forget to update the docs: 
Adapters: https://github.com/insight-platform/Savant/blob/develop/docs/adapters.md
Readme: https://github.com/insight-platform/Savant/
Architecture: https://github.com/insight-platform/Savant/blob/develop/docs/architecture.md

File: docs/adapters.md [hunk 2]
```diff
@@ -14,8 +14,8 @@ The framework can communicate with both source and sink adapters via a protocol
   - [PUB/SUB Explanation](#pub/sub-explanation)
 - [Source Adapters](#source-adapters)
   - [The Image File Source Adapter](#the-image-file-source-adapter)
-- [The Video File Source Adapter](#the-video-file-source-adapter)
-- [Example](#example)
+  - [The Video File Source Adapter](#the-video-file-source-adapter)
+  - [The Looped Video File Source Adapter](#the-looped-video-file-source-adapter)
```
Comment (bwsw) [inline docs/adapters.md:18]: Please change to Loop Video File Source Adapter, don't use the 'looped' eveywhere.

Comment (bwsw) [reply docs/adapters.md:18]: We could use Cycled Video, but Looped Video is a weird thing.

**Review by bwsw — CHANGES_REQUESTED** [review]

File: docs/adapters.md (hunk 2 above)
Comment (bwsw) [reply docs/adapters.md:18]: Even better to change to "Video Loop Source Adapter".

**Review by dorgun — CHANGES_REQUESTED** [review]
Not implemented "Make It possible to configure a random loss rate, the parameter which makes the adapter randomly drop certain frames (needed to stress the module)."

File: adapters/gst/gst_plugins/python/media_files_src_bin.py [hunk 3]
```diff
@@ -334,6 +440,16 @@ def start_next_file(self, file_location: str):
             self.remove(elem)
         self._elements = []
 
+        if self.source.get_factory().get_name() != 'filesrc':
```
[the remark this answers is not in the document]
Comment (tomskikh) [reply adapters/gst/gst_plugins/python/media_files_src_bin.py:443]: This needed to change "souphttpsrc" to "filesrc".

File: adapters/gst/gst_plugins/python/media_files_src_bin.py [hunk 4]
```diff
@@ -342,7 +458,10 @@ def start_next_file(self, file_location: str):
         return False
 
     def pop_next_location(self):
-        return self.pending_locations.pop(0)
+        if self.loop_file and len(self.pending_locations) == 1:
```
[the remark this answers is not in the document]
Comment (tomskikh) [reply adapters/gst/gst_plugins/python/media_files_src_bin.py:461]: Yes. When the source is HTTP URL then `self.pending_locations` contains that URL and downloaded file location.

**Comment by tomskikh** [comment]
> Not implemented "Make It possible to configure a random loss rate, the parameter which makes the adapter randomly drop certain frames (needed to stress the module)."

@dorgun Now it's implemented.

**Review by dorgun — APPROVED** [review]

**Review by bwsw — APPROVED** [review]

**Pull request merged**